driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -243,7 +246,8 @@ private static String extractMessage(Throwable t) {
                         // Testing for a specific string is a tad fragile but well, we don't have much choice
                         // C* 2.1 reports a server error instead of protocol error, see CASSANDRA-9451
                         if ((error.code == ExceptionCode.PROTOCOL_ERROR || error.code == ExceptionCode.SERVER_ERROR) &&
-                                error.message.contains("Invalid or unsupported protocol version"))
+                                error.message.contains("Invalid or unsupported protocol version") ||
```

Comment:
I think the precedence of boolean operators would require parentheses here:
```
(
    error.code == ExceptionCode.PROTOCOL_ERROR || 
    error.code == ExceptionCode.SERVER_ERROR
) && (
    error.message.contains("Invalid or unsupported protocol version") || 
    error.message.contains("Beta version of the protocol used")
)
```
Or is it enough to detect the `"Beta version of the protocol used"` message, regardless of the error code?